include/chimera/frontend_action.h

```diff
@@ -14,10 +19,39 @@ namespace chimera
 class FrontendAction : public clang::ASTFrontendAction
 {
 public:
+    // Overrides the constructor in order to receive ChimeraConfiguration.
+    FrontendAction(const chimera::Configuration &config);
+
     std::unique_ptr<clang::ASTConsumer> CreateASTConsumer(
         clang::CompilerInstance &CI, clang::StringRef file) override;
+
+protected:
+    const chimera::Configuration &config_;
+};
+
+// Create a custom action factory that forwards the ChimeraConfiguration.
+class ChimeraFrontendActionFactory
+  : public clang::tooling::FrontendActionFactory
+{
+public:
+    ChimeraFrontendActionFactory(const chimera::Configuration &config);
+
+#if LLVM_VERSION_AT_LEAST(8, 0, 0)
+    std::unique_ptr<clang::FrontendAction> create() override;
+#else
+    clang::FrontendAction *create() override;
+#endif
+
+protected:
+    const chimera::Configuration &config_;
 };
 
+/**
+ * Custom frontend factory that forwards a ChimeraConfiguration.
+ */
+std::unique_ptr<clang::tooling::FrontendActionFactory> newFrontendActionFactory(
+    const chimera::Configuration &config);
+
```

Comment:
This approach is summarized in this comment thread requesting this functionality be added to LLVM.  It doesn't look like it was added though :(

http://lists.llvm.org/pipermail/cfe-dev/2014-December/040607.html